src/PokemonTrainer.java

```diff
@@ -0,0 +1,26 @@
+import java.util.List;
+
+public class PokemonTrainer {
+    String  name;
```

Comment:
De name variabele mag final zijn, omdat je het 1x in de constructor zet en je iemands naam doorgaans niet veranderd. 
Final variabelen kunnen geen setter hebben, omdat je maar 1x mag zetten (in de cunstructor). Maar jou setter was toch al effectief onbruikbaar.